sysvar/src/epoch_schedule.rs

```diff
@@ -119,17 +119,124 @@
 //! #
 //! # Ok::<(), anyhow::Error>(())
 //! ```
+use crate::Sysvar;
 #[cfg(feature = "bincode")]
 use crate::SysvarSerialize;
-use crate::{impl_sysvar_get, Sysvar};
 pub use {
     solana_epoch_schedule::EpochSchedule,
     solana_sdk_ids::sysvar::epoch_schedule::{check_id, id, ID},
 };
 
+/// Pod (Plain Old Data) representation of [`EpochSchedule`] with no padding.
+///
+/// This type can be safely loaded via `sol_get_sysvar` without undefined behavior.
+/// Provides performant zero-copy accessors as an alternative to the `EpochSchedule` type.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct PodEpochSchedule {
+    slots_per_epoch: [u8; 8],
+    leader_schedule_slot_offset: [u8; 8],
+    warmup: u8,
+    first_normal_epoch: [u8; 8],
+    first_normal_slot: [u8; 8],
+}
+
+const POD_EPOCH_SCHEDULE_SIZE: usize = 33;
+const _: () = assert!(core::mem::size_of::<PodEpochSchedule>() == POD_EPOCH_SCHEDULE_SIZE);
+
+impl PodEpochSchedule {
+    /// Fetch the sysvar data using the `sol_get_sysvar` syscall.
+    /// This provides an alternative to `EpochSchedule` which provides zero-copy accessors.
+    pub fn fetch() -> Result<Self, solana_program_error::ProgramError> {
+        let mut pod = core::mem::MaybeUninit::<Self>::uninit();
+        // Safety: `get_sysvar_unchecked` will initialize `pod` with the sysvar data,
+        // and error if unsuccessful.
+        unsafe {
+            crate::get_sysvar_unchecked(
+                pod.as_mut_ptr() as *mut u8,
+                (&id()) as *const _ as *const u8,
+                0,
+                POD_EPOCH_SCHEDULE_SIZE as u64,
+            )?;
+            Ok(pod.assume_init())
+        }
+    }
+
+    pub fn slots_per_epoch(&self) -> u64 {
+        u64::from_le_bytes(self.slots_per_epoch)
+    }
+
+    pub fn leader_schedule_slot_offset(&self) -> u64 {
+        u64::from_le_bytes(self.leader_schedule_slot_offset)
+    }
+
+    pub fn warmup(&self) -> bool {
+        self.warmup != 0
+    }
```

Comment:
This works, but I am not sure whether we need to be more specific and really test the only valid values.
```suggestion
    pub fn warmup(&self) -> Result<bool, ProgramError> {
        match self.warmup {
            0 => Ok(false),
            1 => Ok(true),
            _ => Err(ProgramError::InvalidAccountData),
        }
    }
```
The other alternative is to return the `u8` value.
```suggestion
    pub fn warmup(&self) -> u8 {
        self.warmup
    }
```